pkg/cloudprovider/suite_test.go

```diff
@@ -722,28 +722,87 @@ var _ = Describe("CloudProvider", func() {
 			Expect(isDrifted).To(Equal(cloudprovider.AMIDrift))
 		})
 		Context("Static Drift Detection", func() {
-			DescribeTable("should return drifted if the spec is updated",
+			BeforeEach(func() {
+				nodeClass = &v1beta1.EC2NodeClass{
+					ObjectMeta: nodeClass.ObjectMeta,
+					Spec: v1beta1.EC2NodeClassSpec{
+						SubnetSelectorTerms:        nodeClass.Spec.SubnetSelectorTerms,
+						SecurityGroupSelectorTerms: nodeClass.Spec.SecurityGroupSelectorTerms,
+						Role:                       nodeClass.Spec.Role,
+						UserData:                   lo.ToPtr("Fake Userdata"),
+						Tags: map[string]string{
+							"fakeKey": "fakeValue",
+						},
+						Context:                  lo.ToPtr("fake-context"),
+						DetailedMonitoring:       lo.ToPtr(false),
+						AMIFamily:                lo.ToPtr(v1beta1.AMIFamilyAL2023),
+						AssociatePublicIPAddress: lo.ToPtr(false),
+						MetadataOptions: &v1beta1.MetadataOptions{
+							HTTPEndpoint:            lo.ToPtr("disabled"),
+							HTTPProtocolIPv6:        lo.ToPtr("disabled"),
+							HTTPPutResponseHopLimit: lo.ToPtr(int64(1)),
+							HTTPTokens:              lo.ToPtr("optional"),
+						},
+						BlockDeviceMappings: []*v1beta1.BlockDeviceMapping{
+							{
+								DeviceName: lo.ToPtr("fakeName"),
+								RootVolume: false,
+								EBS: &v1beta1.BlockDevice{
+									DeleteOnTermination: lo.ToPtr(false),
+									Encrypted:           lo.ToPtr(false),
+									IOPS:                lo.ToPtr(int64(0)),
+									KMSKeyID:            lo.ToPtr("fakeKMSKeyID"),
+									SnapshotID:          lo.ToPtr("fakeSnapshot"),
+									Throughput:          lo.ToPtr(int64(0)),
+									VolumeSize:          resource.NewScaledQuantity(2, resource.Giga),
+									VolumeType:          lo.ToPtr("standard"),
+								},
+							},
+						},
+					},
+				}
+				nodeClass.Annotations = lo.Assign(nodeClass.Annotations, map[string]string{v1beta1.AnnotationEC2NodeClassHash: nodeClass.Hash()})
+				nodeClaim.Annotations = lo.Assign(nodeClaim.Annotations, map[string]string{v1beta1.AnnotationEC2NodeClassHash: nodeClass.Hash()})
+			})
+			DescribeTable("should return drifted if a statically drifted EC2NodeClass.Spec field is updated",
 				func(changes v1beta1.EC2NodeClass) {
 					ExpectApplied(ctx, env.Client, nodePool, nodeClass)
 					isDrifted, err := cloudProvider.IsDrifted(ctx, nodeClaim)
 					Expect(err).NotTo(HaveOccurred())
 					Expect(isDrifted).To(BeEmpty())
 
-					Expect(mergo.Merge(nodeClass, changes, mergo.WithOverride))
+					Expect(mergo.Merge(nodeClass, changes, mergo.WithOverride, mergo.WithSliceDeepCopy))
```

Comment:
What's the slice deepcopy for? You're modifying slice information and that's somehow affecting the original?

Reply:
Yes, `blockDeviceMapping` is the slice of concern here. When we do a merge the whole since is getting changed, as opposed to the individual field we care about